File: src/main/java/core/procedures/interop/BooleanType.java
package core.procedures.interop;

import core.procedures.AFn;
import core.procedures.FnArgsBuilder;
import core.utils.Utils;

public final class BooleanType extends AFn {

  public BooleanType() {
    super(new FnArgsBuilder().min(1).max(1).build());
  }

  @Override
  public boolean isPure() {
    return true;
  }

  @Override
  public String getName() {
    return "boolean";
  }

  @Override
  public Boolean apply1(Object arg) {
    /* Have to box it */
    if (Utils.toBoolean(arg)) {
      return Boolean.TRUE;
    } else {
      return Boolean.FALSE;
    }
  }
}

File: src/main/java/core/procedures/interop/BooleanType.kt
package core.procedures.interop

import core.procedures.AFn
import core.procedures.Arity.Exactly
import core.utils.Utils

class BooleanType : AFn<Any?, Boolean>(name = "boolean", isPure = true, arity = Exactly(1)) {

    override operator fun invoke(arg: Any?) = Utils.toBoolean(arg)
}
